dfastmi/io/RiversObject.py

```diff
@@ -94,24 +94,24 @@ def _read_rivers_file(self, filename):
         self._parse_branches(config)
 
         # parse reaches and discharge locations
-        self._parse_reaches(config, iversion)
+        self._parse_reaches(config, self.version)
 
         # call the specific reader for the file version
-        if iversion == 1:
+        if self.version == Version("1"):
             self._read_rivers_legacy(river_data)
 
-        else:  # iversion == 2
+        else:
             self._read_rivers(river_data)
 
-    def _parse_reaches(self, config: configparser.ConfigParser, iversion):
+    def _parse_reaches(self, config: configparser.ConfigParser, version: Version):
```

Comment:
You can leave passing in the version argument out of this method call, since you pass in a class property you can just use it in that method. e.g. self.version.

Reply:
@ralph-peelen-deltares  will create PR.

Reply:
PR:#108 Discussed with Bert.